src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,178 @@
+  const authTypes = [API_KEY, HTTP, OAUTH2, OPENID_CONNECT];
+  const errors = [];
+  const warnings = [];
+  const securitySchemes =
+    resolvedSpec.components && resolvedSpec.components.securitySchemes;
+
+  for (const key in securitySchemes) {
+    const path = `securitySchemes.${key}`;
+    const security = securitySchemes[key];
+    const type = security.type;
+
+    if (!type) {
+      errors.push({
+        message: 'security scheme is missing required field `type`',
+        path
+      });
+    } else if (authTypes.indexOf(type) === -1) {
+      errors.push({
+        message:
+          '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`',
+        path: path + '.type'
+      });
+    } else if (type === API_KEY) {
+      //apiKey validation
+      const authIn = security.in;
+      if (!authIn || !['query', 'header', 'cookie'].includes(authIn)) {
+        errors.push({
+          message:
+            "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+          path: path + '.in'
+        });
+      }
+      if (!security.name) {
+        errors.push({
+          message:
+            "apiKey authorization must have required 'name' string property. The name of the header or query property to be used.",
+          path
+        });
+      }
+    }
+    // oauth2 validation
+    else if (type === OAUTH2) {
+      const flows = security.flows;
+
+      if (!flows) {
+        errors.push({
+          message: "oauth2 authorization must have required 'flows' property",
+          path
+        });
+      } else if (flows.authorizationCode && !flows.authorizationCode.tokenUrl) {
+        errors.push({
+          message:
+            "flow must have required 'tokenUrl' property if type is `authorizationCode`",
+          path: path + '.flows.authorizationCode'
+        });
+      } else if (flows.password && !flows.password.tokenUrl) {
+        errors.push({
+          message:
+            "flow must have required 'tokenUrl' property if type is `password`",
+          path: path + '.flows.password'
+        });
+      } else if (flows.clientCredentials && !flows.clientCredentials.tokenUrl) {
+        errors.push({
+          message:
+            "flow must have required 'tokenUrl' property if type is  `clientCredentials`",
+          path: path + '.flows.clientCredentials'
+        });
+      } else if (
+        !flows.implicit &&
+        !flows.authorizationCode &&
+        !flows.password &&
+        !flows.clientCredentials
+      ) {
+        errors.push({
+          message:
+            "oauth2 authorization `flows` must have one of the following properties: 'implicit', 'password', 'clientCredentials' or 'authorizationCode'",
+          path: path + '.flows'
+        });
+      } else if (flows.implicit) {
+        const authorizationUrl = flows.implicit.authorizationUrl;
+        if (!flows.implicit.scopes || !authorizationUrl) {
```

Comment:
You don't need this `if` statement. You are checking for `if (!authorizationUrl)` and `if (!flows.implicit.scopes)` below, there is no need to wrap that in a check for either of them